### DISCUSSION TITLE
How to stop logging for micro-sd

### DISCUSSION THREAD

**Discussion by easy98** [bitcraze/discussions#25] [Q&A] [answered]
Hi,

When I run `rosparam set /cf1/usd/logging 0` to stop logging for micro-sd card, it doesn't work and the micro-sd card is still logging. So how can I externally set the parameter `usd.logging` to 0 and stop logging? Thanks.

**Comment by krichardsson** [comment]
I'm not sure what your ROS setup looks like, but the correct way to stop the SD-card logging is indeed to set the `usd.logging` parameter to 0 

You can read more under the "Usage" tab here https://www.bitcraze.io/products/micro-sd-card-deck/

Maybe you can use the Crazyflie client to set the parameter to figure out if it is a ROS problem or not?

**Comment by whoenig** [comment, marked as answer]
ROS (crayzflie_ros and Crazyswarm) require a service call to upload the changed parameters.

**Reply by Rwik2000** [reply]
Can you please summarise what all needs to be done for logging to start and stop in the uSD deck? From my understanding:

1.  Change the config.txt in the crazyflie_firmware and upload the firmware to the crazyflie.
2.  Running `rosparam set /cf1/usd/logging 1` to toggle the usd.logging variable in the firmware and thus start logging.
3.  Use `rosservice call` to upload the changed parameter (Can you please give more details as to how to do it?)
4.  Run `rosparam set /cf1/usd/logging 0` to stop logging

**Comment by easy98** [comment]
Thanks for your reply. Now I use the client to stop it and it works!
